### ISSUE TITLE
[Bug] Diff stats silently corrupted by wrong tensor shape in distributed compute_standardization_stats.py

### ISSUE THREAD

**Issue by RajdeepKushwaha5** [mllam/neural-lam#413]
## Description

Running `compute_standardization_stats.py --distributed` with `world_size > 1` silently writes incorrect `diff_mean.pt` and `diff_std.pt` files with no
exception raised.

## Root cause

After gathering, `diff_means_gathered` has shape `(total_window_samples, d_features)`. The original code uses a Python list comprehension to select rows:

```python
diff_means = [diff_means_gathered[i] for i in original_indices]
```

Each `diff_means_gathered[i]` is a 1-D `(d_features,)` tensor (a single row). `torch.cat(dim=0)` on a list of 1-D tensors concatenates along the *only*
dimension, producing a flat `(N * d_features,)` tensor instead of the expected `(N, d_features)` 2-D tensor.

`save_stats` then calls `torch.mean(tensor, dim=0)` on this 1-D tensor, which returns a *scalar* mean of all values instead of the `(d_features,)`
per-variable mean. The incorrect scalar is written to disk without any error.

## Impact

All per-variable diff statistics (`diff_mean`, `diff_std`) computed in distributed mode are wrong — they collapse to a single scalar that is
broadcast as the mean/std for every variable.

## Fix

Use PyTorch fancy indexing to preserve both dimensions:

```python
diff_means = [diff_means_gathered[original_indices]]  # (N, d_features)
```

`torch.cat` on a single `(N, d_features)` tensor is a no-op on `dim=0`,
giving the correct `(N, d_features)` shape for `save_stats`.

Fixed in PR #411.
